cognite_toolkit/_cdf_tk/commands/deploy.py

```diff
@@ -192,32 +186,29 @@ def execute(
     def deploy_resources(
         self,
         loader: Loader,
-        path: Path,
         ToolGlobals: CDFToolConfig,
         dry_run: bool = False,
         has_done_drop: bool = False,
         has_dropped_data: bool = False,
         verbose: bool = False,
     ) -> DeployResult | None:
         if isinstance(loader, ResourceLoader):
-            return self._deploy_resources(loader, path, ToolGlobals, dry_run, has_done_drop, has_dropped_data, verbose)
+            return self._deploy_resources(loader, ToolGlobals, dry_run, has_done_drop, has_dropped_data, verbose)
         elif isinstance(loader, DataLoader):
-            return self._deploy_data(loader, path, ToolGlobals, dry_run, verbose)
+            return self._deploy_data(loader, ToolGlobals, dry_run, verbose)
         else:
             raise ValueError(f"Unsupported loader type {type(loader)}.")
 
     def _deploy_resources(
         self,
         loader: ResourceLoader,
-        path: Path,
         ToolGlobals: CDFToolConfig,
         dry_run: bool = False,
         has_done_drop: bool = False,
         has_dropped_data: bool = False,
         verbose: bool = False,
     ) -> ResourceDeployResult | None:
-        loader.build_path = path
```

Comment:
This looks strange, you are setting an attribute on an input argument (mutating an input argument) with another input argument. 

This is one of the motivation for removing the `path` (which is the build path of the resource`, which is set in the init of the loader.